Pirat/Model/Api/Resource/Item.cs

```diff
@@ -51,6 +51,14 @@ public override bool Equals(object obj)
 
         public bool Equals(Item other)
         {
+            if (address != null && other != null && !address.Equals(other.address))
+            {
+                return false;
+            }
+            if (address == null && other.address != null)
+            {
+                return false;
+            }
```

Comment:
What's about the case were `address != null` and `other.address == null`?

Also, why are you doing this anyway? Why can't you use `Equals(address, other.address)`?